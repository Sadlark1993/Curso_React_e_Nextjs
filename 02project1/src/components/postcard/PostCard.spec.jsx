/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-debugging-utils */

import { render, screen } from "@testing-library/react"
import { PostCard } from "."

//crating mock data
const mockData = {
  title: 'A Random Title',
  body: 'Some body... that song from Shrek.',
  photo: './img/img.png',
}


describe('<PostCard />', ()=>{
  it('should render PostCard correctly',()=>{
    //const {debug} = render(<PostCard {...mockData} />);

    //gonna print the output html in terminal.
    //debug();
  
    render(<PostCard {...mockData} />);
    expect(screen.getByRole('img', {name: mockData.title})).toHaveAttribute('src', mockData.photo);
    expect(screen.getByRole('heading', {name: mockData.title})).toBeInTheDocument();
    expect(screen.getByText(/Shrek/i)).toBeInTheDocument();
  });

  it('should match snapshot', ()=>{
    const {container} = render(<PostCard {...mockData} />);
    expect(container.firstChild).toMatchSnapshot();
  })
})