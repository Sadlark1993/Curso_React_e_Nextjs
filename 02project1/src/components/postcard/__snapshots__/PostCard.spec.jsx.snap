// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="A Random Title"
    src="./img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      A Random Title
    </h2>
    <p>
      Some body... that song from Shrek.
    </p>
  </div>
</div>
`;
